src/networks.jl

```diff
@@ -14,6 +14,7 @@ struct RegisterNet
 end
 
 function RegisterNet(graph::SimpleGraph, registers, vertex_metadata, edge_metadata, directed_edge_metadata)
+    glcnt[] = 0 # set the global counter of `guid`s to zero whenever a new network is initialized
```

Comment:
That seems extremely dangerous if someone wants to run multiple simulations, e.g. on separate threads. We should not do this.